kiwi/filesystem/base.py

```diff
@@ -43,7 +47,10 @@ class FileSystemBase:
     :param string root_dir: root directory path name
     :param dict custom_args: custom filesystem arguments
     """
-    def __init__(self, device_provider, root_dir=None, custom_args=None):
+    def __init__(
+        self, device_provider: DeviceProvider,
+        root_dir: str = None, custom_args: Dict = None
```

Comment:
can we set the types of `Dict` ? or they are not known ?

Reply:
Well I did not do that because it is not in a very strict format. This dict format should be something like `Dict[str, List or Dict or str]` I did not even check if it is possible to express it. I thought it already enough stating it is a dict.

See there is an example of a custom_args dict in `post_init` method few lines below.